sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -399,66 +489,73 @@ export class ManagementClient extends LinkEntity {
+          if (associatedLinkName) {
+            request.application_properties![Constants.associatedLinkName] = associatedLinkName;
+          }
+          request.application_properties![Constants.trackingId] = generate_uuid();
+          log.mgmt(
+            "[%s] Peek by sequence number request body: %O.",
+            this._context.namespace.connectionId,
+            request.body
+          );
 
-      const result = await this._mgmtReqResLink!.sendRequest(request);
-      if (result.application_properties!.statusCode !== 204) {
-        const messages = result.body.messages as { message: Buffer }[];
-        for (const msg of messages) {
-          const decodedMessage = RheaMessageUtil.decode(msg.message);
-          const message = fromAmqpMessage(decodedMessage as any);
-          message.body = this._context.namespace.dataTransformer.decode(message.body);
-          messageList.push(message);
-          this._lastPeekedSequenceNumber = message.sequenceNumber!;
+          const result = await this._acquireLockAndSendRequest(request, retryTimeoutInMs, {
+            abortSignal: undefined,
+            requestName: undefined
+          });
+          if (result.application_properties!.statusCode !== 204) {
+            const messages = result.body.messages as { message: Buffer }[];
+            for (const msg of messages) {
+              const decodedMessage = RheaMessageUtil.decode(msg.message);
+              const message = fromAmqpMessage(decodedMessage as any);
+              message.body = this._context.namespace.dataTransformer.decode(message.body);
+              messageList.push(message);
+              this._lastPeekedSequenceNumber = message.sequenceNumber!;
+            }
+          }
+          resolve(messageList);
+        } catch (err) {
+          const error = translate(err) as MessagingError;
+          log.error(
+            "An error occurred while sending the request to peek messages to " +
+              "$management endpoint: %O",
+            error
+          );
+          // statusCode == 404 then do not throw
+          if (error.code !== ConditionErrorNameMapper["com.microsoft:message-not-found"]) {
+            reject(error);
+          }
         }
-      }
-    } catch (err) {
-      const error = translate(err) as MessagingError;
-      log.error(
-        "An error occurred while sending the request to peek messages to " +
-          "$management endpoint: %O",
-        error
-      );
-      // statusCode == 404 then do not throw
-      if (error.code !== ConditionErrorNameMapper["com.microsoft:message-not-found"]) {
-        throw error;
-      }
-    }
-    return messageList;
+      });
+    const config: RetryConfig<ReceivedMessage[]> = {
+      operation: peekBySequenceNumberOperationPromise,
+      connectionId: this._context.namespace.connectionId,
+      operationType: RetryOperationType.management,
+      retryOptions: retryOptions
+    };
+    return retry<ReceivedMessage[]>(config);
```

Comment:
What is the reason to choose to call `retry()` from each of the methods here instead of doing it in a single place like we did in Event Hubs?

Reply:
Right now, the retry() is moved to an even higher level as I mentioned before - https://github.com/Azure/azure-sdk-for-js/pull/7972#discussion_r400072487.
retry() is being called from the methods in the Receiver, SessionReceiver and SubscriptionRuleManager classes.

For instance, any method in the SessionReceiver creates a session before calling the corresponding method from managementClient. 
In this case, `creating-message-session` involves `initialization`, `negotiating claims`, etc, `timeout-per-retry-instance` is being shared by all the calls from creating the session to the managementLink's sendRequest. 

This makes more sense than pushing the retry() call to the lowest layer in the Hierarchy.

Reply:
> What is the reason to choose to call retry() from each of the methods here instead of doing it in a single place like we did in Event Hubs?

Looking at the event-hubs code after your comment...
![image](https://user-images.githubusercontent.com/10452642/77902192-a842ba80-7235-11ea-8dc4-d1b70c2f76e8.png)
From the above screenshot, `retry()` call was added in the `_makeManagementRequest` method only, the `getSecurityToken()` method is not guarded with any retries(might be the case that this was intentional).

Reply:
Whether retries are to be performed on the `getSecurityToken()` is dependent on what the best practices are on the `getToken()` method of a `TokenCredential`. I would guess that the credential uses retry policies internally. 

Also, on second thoughts, I believe the call to `getSecurityToken()` should go inside `_makeManagementRequst()`. Its in the latter that we do claim negotiation for the link if it doesnt exist. The token used for claim negotiation can be cached and re-used to update the "request". We shouldn't need to call `getSecurityToken()` if the link is already established.

This is of course tangential to this PR :)
cc @chradek